adagrams/game.py

```diff
@@ -1,11 +1,193 @@
+
+        # get a random index from the letters list
+        random_index = random.randint(0, len(letters) -1)
+
+        # get the random letter by using the random index
+        random_letter = letters[random_index]
+        # get the letter counts from the copy of letter pool, everytime draw a letter, the value update -1
+        num = user_pool[random_letter]
+        user_pool[random_letter] = num - 1
+
+        #if no remaining letter in copy of letter pool, skip, and re-draw
+        if user_pool[random_letter] <= 0:
+            continue
+        else:
+            hand.append(random_letter)
+
+    return hand
+    
+'''
+    #use random.choice()
+    # when the user has less 10 letter on hand, 
+    # continue to drawn from the user pool, store in the list
+    # if the remaining letter in user pool is 0,
+    # jump to drawn a new letter.
+    while len(hand) < 10:
+        
+        letter = random.choice(list(user_pool.keys()))
+        
+        num = user_pool[letter]
+
+        user_pool[letter] = num - 1;
+        
+        if user_pool[letter] <= 0:
+            continue
+        else:
+            hand.append(letter)
+'''
+    
+    
+    
 
 def uses_available_letters(word, letter_bank):
-    pass
+    
+    #initialize an empty dictinary to store the frequency of each letter
+    frequency = {}
+
+    #loop through each letter from the letter_bank,
+    #if the letter in frequency{}, value +1
+    #if not, add the letter to frequency{} and value 1
+    for letter in letter_bank:
+        if letter in frequency:
+            frequency[letter] += 1
+        else:
+            frequency[letter] = 1  
+
+    #check if word in letter_bank()
+    #ignore the case sensitive
+    #if the letter in the word,
+    #and the freqency of the letter in word is lees than frequency of letter in letter bank, return True
+    #else return False
+    for letter in word.upper():
+        if letter in frequency.keys():
+            if frequency[letter] > 0:
+                frequency[letter] -= 1
+            else:
+                return False
+        else:
+            return False
+        
+    return True
+
+    '''
+    #using Counter() method
+    conuts = dict(Counter(letter_bank))
+    print(conuts)
+
+    #check if letter is in letter bank
+    #
+    for letter in word.upper():
+        if letter in conuts.keys():
+            if conuts[letter] > 0:
+                conuts[letter] -= 1
+            else:
+                return False
+        else:
+            return False
+    return True
+    '''
+
+
+SCORE_CHART = {
+    ('A','E','I','O','U','L','N','R','S','T'):1,
+    ('D','G'): 2,
+    ('B','C','M','P'): 3,
+    ('F','H','V','W','Y'): 4,
+    ('K'):5,
+    ('J','X'): 8,
+    ('Q','Z'): 10
+}
+
 
 def score_word(word):
-    pass
+    
+    #initiate a varaible to store the score
+    score = 0
+
+    #if the length of word is 7,8,9,10, the word gets an additional 8 points 
+    if len(word) > 6:
+        score += 8
```

Comment:
I should have called this out earlier. Prefer the literal integers to be referenced by variables to make your code more self-documenting: 

```python
BONUS_LENGTH = 7
BONUS_POINTS = 8

if len(word) >= BONUS_LENGTH:
        total_score = total_score + BONUS_POINTS
```

Reply:
While a hand can't be more than 10 letters and therefore a word can't be more than 10 letters, you could make this check more explicit and conform to the instructions in the README more by having the check on line 154 be: `if 7 <= len(word) <= 10:`